trace-utils/src/send_data.rs

```diff
@@ -151,79 +302,508 @@ impl SendData {
             req = req.header(*key, value);
         }
 
-        async fn send_request(
-            req: HttpRequestBuilder,
-            payload: Vec<u8>,
-        ) -> Result<Response<Body>, SendRequestError> {
-            let req = req
-                .body(Body::from(payload))
-                .map_err(|e| SendRequestError::Any(anyhow!(e)))?;
-
-            Client::builder()
-                .build(connector::Connector::default())
-                .request(req)
-                .await
-                .map_err(SendRequestError::Hyper)
-        }
+        req
+    }
+
+    async fn send_request(
+        &self,
+        req: HttpRequestBuilder,
+        payload: Vec<u8>,
```

Comment:
If this is a `Bytes` instead, then there can be a shallow arc copy in the calling code, since the `Bytes` will be constructed from a `Vec<u8>`